frame/offences/benchmarking/src/lib.rs

```diff
@@ -216,56 +216,63 @@ fn make_offenders_im_online<T: Config>(
 	Ok((id_tuples, offenders))
 }
 
-#[cfg(test)]
-fn check_events<T: Config, I: Iterator<Item = <T as SystemConfig>::RuntimeEvent>>(expected: I) {
+fn check_events<
+	T: Config,
+	I: Iterator<Item = Item>,
+	Item: sp_std::borrow::Borrow<<T as SystemConfig>::RuntimeEvent> + sp_std::fmt::Debug,
+>(
+	expected: I,
+) {
 	let events = System::<T>::events()
 		.into_iter()
 		.map(|frame_system::EventRecord { event, .. }| event)
 		.collect::<Vec<_>>();
 	let expected = expected.collect::<Vec<_>>();
 
-	fn pretty<D: std::fmt::Debug>(header: &str, ev: &[D], offset: usize) {
-		println!("{}", header);
+	fn pretty<D: sp_std::fmt::Debug>(header: &str, ev: &[D], offset: usize) {
+		log::info!("{}", header);
 		for (idx, ev) in ev.iter().enumerate() {
-			println!("\t[{:04}] {:?}", idx + offset, ev);
+			log::info!("\t[{:04}] {:?}", idx + offset, ev);
 		}
 	}
-	fn print_events<D: std::fmt::Debug>(idx: usize, events: &[D], expected: &[D]) {
+	fn print_events<D: sp_std::fmt::Debug, E: sp_std::fmt::Debug>(
+		idx: usize,
+		events: &[D],
+		expected: &[E],
+	) {
 		let window = 10;
 		let start = idx.saturating_sub(window / 2);
 		let end_got = (idx + window / 2).min(events.len());
 		pretty("Got(window):", &events[start..end_got], start);
 		let end_expected = (idx + window / 2).min(expected.len());
 		pretty("Expected(window):", &expected[start..end_expected], start);
-		println!("---------------");
+		log::info!("---------------");
 		let start_got = events.len().saturating_sub(window);
 		pretty("Got(end):", &events[start_got..], start_got);
 		let start_expected = expected.len().saturating_sub(window);
 		pretty("Expected(end):", &expected[start_expected..], start_expected);
 	}
-	let events_copy = events.clone();
-	let expected_copy = expected.clone();
-
-	for (idx, (a, b)) in events.into_iter().zip(expected).enumerate() {
-		if a != b {
-			print_events(idx, &events_copy, &expected_copy);
-			println!("Mismatch at: {}", idx);
-			println!("     Got: {:?}", b);
-			println!("Expected: {:?}", a);
-			if events_copy.len() != expected_copy.len() {
-				println!(
+
+	for (idx, (a, b)) in events.iter().zip(expected.iter()).enumerate() {
```

Comment:
Could also have been done with the new `StorageStreamIter`, or?

Reply:
Yes good idea!